src/molecules/modal/Modal.js

```diff
@@ -1,85 +1,119 @@
 // @vue/component
 export default {
   props: {
-    name: {
-      type: String,
-      required: true
-    },
+    /**
+     * Modal heading text
+     */
     heading: {
       type: String,
       default: null
     },
+    /**
+     * Modal visibility indicator
+     */
+    visible: {
+      type: Boolean,
+      default: false
+    },
+    /**
+     * Indicate if clicking outside modal can emit `close` action
+     */
     closeOnBackgroundClick: {
       type: Boolean,
       default: true
     },
+    /**
+     * Indicate if Esc key can emit `close` action
+     */
     closeOnEsc: {
       type: Boolean,
       default: true
     },
-    maxWidth: {
-      type: Number,
-      default: null
+    /**
+     * Modal main transition effect name
+     */
+    modalMainTransition: {
+      type: String,
+      default: 'fade'
+    },
+    /**
+     * Modal content transition effect name
+     */
+    modalContentTransition: {
+      type: String,
+      default: 'slide-down'
     },
+    /**
+     * Modal aria-labelledby text
+     */
     ariaLabelledby: {
       type: String,
       default: null
     },
+    /**
+     * Modal aria-describedby text
+     */
     ariaDescribedby: {
       type: String,
       default: null
     },
+    /**
+     * Indicate if close button is available
+     */
     closeButton: {
       type: Boolean,
       default: true
     },
-    blank: {
-      type: Boolean,
-      default: false
+    /**
+     * Modal close button aria-label text
+     */
+    closeButtonAriaLabel: {
+      type: String,
+      default: null
     }
   },
   data () {
     return {
-      visibility: false,
-      trigger: null,
-      ariaHidden: 'true'
+      trigger: null
     }
   },
   computed: {
-    getMaxWidth () {
-      return this.maxWidth !== null ? `max-width: ${this.maxWidth}px;` : null
+    ariaHidden () {
+      return this.visible ? 'false' : 'true'
     }
   },
-  methods: {
-    show (name) {
-      if (name === this.name) {
+  watch: {
+    visible (val) {
+      if (val && typeof window !== 'undefined') {
         this.trigger = document.activeElement
-        this.toggleParams(true)
         this.$nextTick(() => this.$refs.modal.focus())
       }
-    },
-    hide (name) {
-      if (name === this.name) {
-        this.toggleParams(false)
+    }
+  },
+  methods: {
+    close () {
+      /**
+       * Trigger close action
+       * @type {Event}
+       */
+      this.$emit('close')
+
+      if (this.trigger) {
         this.$nextTick(() => this.trigger.focus())
       }
     },
-    toggleParams (state) {
-      this.ariaHidden = state ? 'false' : 'true'
-      this.visibility = state
-    },
     handleBackgroundClick () {
       if (this.closeOnBackgroundClick) {
-        this.hide(this.name)
+        this.close()
       }
     },
     handleKeydown (event) {
-      if (this.visibility) {
+      if (this.visible) {
         switch (event.key) {
           case 'Esc': // IE/Edge specific value
```

Comment:
this case is empty, is it ok?

Reply:
in other modules we use `KeyCodes`. IMO we should follow the same approach

Reply:
It's empty because we want to use one action for two cases. Ok then, `keyCode` used instead `key`, as far as I checked it should be same kode over ie and other modern browsers.